modules/end-user-portal/src/Pages/Apply/SoftwareSupplier.cshtml.cs

```diff
@@ -0,0 +1,113 @@
+            SelectedSoftwareSupplierNameId = selectedSoftwareSupplierNameId.SoftwareSupplierId;
+            SoftwareSupplierProductList = _tempDataProviderService.GetItem<List<SoftwareSupplierProductResult>>("SelectedSoftwareSupplierProduct");
+            DisplaySoftwareSupplierProducts = SoftwareSupplierProductList != null;
+        }
+    }
+
+    protected async Task GetSoftwareSupplierNameList()
+    {
+        var suppliers = await _supplierService.GetSoftwareSuppliersAsync();
+        _tempDataProviderService.PutItem("SoftwareSupplierNameList", suppliers);
+    }
+
+    public IActionResult OnPostCheckSupplierProductsAsync()
+    {        
+        if (!ModelState.IsValid)
+        {
+            return Page();
+        }
+        CheckTempData();        
+        DisplaySoftwareSupplierProducts = true;
+        LoadSoftwareSupplierProducts(SelectedSoftwareSupplierNameId);
+        SoftwareSupplierProductList = _tempDataProviderService.GetItem<List<SoftwareSupplierProductResult>>("SoftwareSupplierProductList");
+        return Page();
+    }
+
+    public IActionResult OnPostNextAsync()
+    {
+        CheckSoftwareSupplierProductSelection();
+        if (!ModelState.IsValid)
+        {
+            DisplaySoftwareSupplierProducts = true;
+            return Page();
+        }
+
+        CheckTempData();
+
+        _tempDataProviderService.PutItem("SelectedSoftwareSupplierName", SelectedSoftwareSupplier);
+        _tempDataProviderService.PutItem("SelectedSoftwareSupplierProduct", SoftwareSupplierProductList);       
+
+        return RedirectToPage("./Organisation");
+    }
+
+    private void CheckSoftwareSupplierProductSelection()
+    {
+        if(!SoftwareSupplierProductList.Any(x => x.Selected))
+        {
+            ModelState.AddModelError("HasSelectedSoftwareSupplierProducts", ErrorMessageResources.SoftwareSupplierName);
+        }
+    }
+
+    private IActionResult LoadSoftwareSupplierProducts(int selectedSoftwareSupplier)
+    {
+        var supplierProducts = new List<SoftwareSupplierProductResult>() {
+            new SoftwareSupplierProductResult() { SoftwareSupplierProductId = 1, SoftwareSupplierProduct = "Access Record: HTML" },
+            new SoftwareSupplierProductResult() { SoftwareSupplierProductId = 2, SoftwareSupplierProduct = "Access Record: Structured" },
+            new SoftwareSupplierProductResult() { SoftwareSupplierProductId = 3, SoftwareSupplierProduct = "Appointment Management" },
+            new SoftwareSupplierProductResult() { SoftwareSupplierProductId = 4, SoftwareSupplierProduct = "Send Document" }
+        };
+
+        SoftwareSupplierProductList = supplierProducts;
+        _tempDataProviderService.PutItem("SoftwareSupplierProductList", supplierProducts);
+        return Page();
+    }
+
+    private void ClearModelState()
+    {
+        ModelState.ClearValidationState("SelectedSoftwareSupplierNameId");
+    }
+
+    private IActionResult? CheckTempData()
```

Comment:
I'm not sure that this works as expect as the result of CheckTempData is never returned to trigger the redirect. 
Do we need this call on this page as it is the first screen in the journey?

Also why are we CheckTempData? It would be better if this was named CheckSessionExpired() or removed if not needed

Reply:
You're right - fixed now.